### DISCUSSION TITLE
Blank or white Admin page, Rooms and Spoke are fine

### DISCUSSION THREAD

**Discussion by aidanhmiles** [mozilla/hubs#2866] [Hubs Cloud Questions] [answered]
Hello. I previously had a working self-hosted Hub, and I'm sure I broke it by messing with CORS and CSP. At first, we couldn't open rooms, spoke, or the admin, then I tried both restoring and deleting CORS and CSP params from the AWS parameter store (to try to revert to default values), and I rolled over all instances after each config change.

At this point, I can create Rooms and use Spoke, but cannot access the Admin interface (I see a white, blank page). Since the Hubs team is moving over to discussions and I see no previous discussion of the white/blank admin page issue, I wanted to open one in the hopes of cataloguing reasons that anyone might experience this issue.

Other observations:
- Admin page HTML source shows an empty `#ui-root` element
- Admin page HTML source lacks `env:*` or `ret:*` meta tags
- Spoke and Room pages have `env:*` or `ret:*` meta tags
- No network requests appear to fail on any page other than favicon
<img width="507" alt="Screen Shot 2020-08-15 at 11 15 45 AM" src="https://user-images.githubusercontent.com/3161613/90319236-0f15f700-deeb-11ea-9271-5b1bcc0b3b23.png">

- All pages show the same three warnings, 1) cookies and the sameSite atttribute, and 2) trying to connect to an empty hostname from something, both over http and ws. I believe this is new behavior, and suggests to me that a value is missing for a hostname config somewhere. I have no idea where that might be.
<img width="541" alt="Screen Shot 2020-08-15 at 11 16 21 AM" src="https://user-images.githubusercontent.com/3161613/90318965-f7d60a00-dee8-11ea-9083-e530faddc33d.png">

- I've cleared site data multiple times throughout my testing, just in case

- The only params I have left in the parameter store are:
/ita/{hubname}/hubs/deploy/type
/ita/{hubname}/reticulum/resolver/google_poly_api_key
/ita/{hubname}/reticulum/resolver/sketchfab_api_key
/keymaster/{hubname}/guardian-secret-key
/keymaster/{hubname}/janus-and-reticulum-admin
/keymaster/{hubname}/phoenix-key
/keymaster/{hubname}/postgrest-db-secret
/keymaster/{hubname}/reticulum-bot-access-key
/keymaster/{hubname}/reticulum-cookie
/keymaster/{hubname}/reticulum-oauth-token
/keymaster/{hubname}/send-email-password

Let me know if I can provide more details or anything. Any ideas, tips, questions, etc are most appreciated.

**Comment by robinkwilson** [comment, marked as answer]
Thank you for the detailed description of what you're seeing and what you've tried. Helps a lot!

Do you remember the CSP parameter(s) you deleted? 
Did you deploy Cloudflare workers before deleting the parameters in the parameter store?

**Reply by aidanhmiles** [reply]
Want to be as helpful as I can! It's a complex project

1) I was editing `connect-src` only. The only reason I was mucking around in CSP was to allow POST requests to our own API for tracking purposes, and I believe I broke the connect-src directive while doing so.

2) No cloudflare set up as of yet. Should it be? (Or not?)

**Reply by robinkwilson** [reply]
Re. Cloudflare: Just checking. It adds parameter store variables. 

Re: `connect-src`, where did you modify this? In the admin panel, parameter store or somewhere else?

Continuing from: https://github.com/mozilla/hubs/discussions/2863

**Reply by aidanhmiles** [reply]
In the admin panel.

One thing I failed to note before is that while adjusting CSP and CORS, I managed to break spoke, rooms, and room creation entirely but still had access to the admin panel. I'm not sure if this is because I just never closed the admin panel until I got the rooms working again (and therefore never refreshed the page, since I was only submitting that form over and over).

**Reply by aidanhmiles** [reply]
I just noticed this: each main page (room, spoke, admin) has a minified JS entry file in a <script> appended at the end of <body>. *The entry JS file for admin is empty.* That would explain why `#ui-root` is empty but there are no JS errors, React or otherwise. For some reason devtools says 501 bytes are transferred for that file, but when I download it or open it in a new tab, it is a completely empty file. Is that a symptom of anything in specific?

**Reply by robinkwilson** [reply]
**Do you have a custom client?**
**Did you terminate the EC2 instance(s) to restart your instances?** AWS console > EC2 > click on your HC server(s) then "Terminate" instances. New ones will start up immediately after.

If CORS is configured incorrectly, you'll break rooms, spoke, and admin panel pages. They all work again when the CORS setting in the parameter store is removed. The extra CSP configuration, if you put your hub host name, might cause a problem, I haven't tried this myself though.

At this point, it maybe best to redeploy the stack or [Backup and Restore](https://hubs.mozilla.com/docs/hubs-cloud-aws-backup-and-restore.html).

**Reply by robinkwilson** [reply]
I know the backup and restore isn't ideal, I'd like to figure out what's going wrong. But for a quick way to get you back up and running, that is the quickest and most predictable way.

**Reply by aidanhmiles** [reply]
We do have a custom client. As far as I know we haven't touched the admin page part of things, but if this could be the issue I'll try deploying previous versions of our client. I have to assume that client code is why I'm seeing this empty host string where something is attempting to connect to `https://:80`.

And yes, I've tried to be diligent about rolling over instances after every config change. Unfortunately, I never set myself up with 2fa before the admin page disappeared, so I can't ssh in. I feel silly about that one.

I opened a new discussion about backup and restore :) I'm giving that a shot next, later today. Thanks so much for your help.

I'll report back as I make any other observations or if I manage to fix the issue.

**Reply by robinkwilson** [reply]
You have a custom client! 

Run `npm run undeploy` to remove the custom client and revert back to the HC default client. Let's see what happens!

**Reply by robinkwilson** [reply]
Also fetch from the hubs-cloud branch in the hubs repo to ensure you have the latest deployed HC client version if you're going to try deploying the custom client again.

**Reply by aidanhmiles** [reply]
I believe we incorporated the latest from your repo last week. I'll double-check.

There's some (working) code in the custom client that we need for some meetings today, but I'll undeploy and report back this evening. Thank you!

**Reply by robinkwilson** [reply]
Marking as answer for now to get this out of our todo pipeline. Please @ me to reopen.

**Reply by aidanhmiles** [reply]
I can confirm, `npm run undeploy` fixed whatever the issue was. Still unsure what happened. I'll compare our custom client to master  and see if anything useful pops out. Thanks again for your help!
